xarray/core/accessor_str.py

```diff
@@ -846,68 +1085,881 @@ def replace(self, pat, repl, n=-1, case=None, flags=0, regex=True):
+        and returns 3 elements containing the part before the separator,
+        the separator itself, and the part after the separator.
+        If the separator is not found, return 3 elements containing the string itself,
+        followed by two empty strings.
+
+        This is equivalent to :meth:`str.partion`.
+
+        Parameters
+        ----------
+        dim : Hashable or `None`
+            Name for the dimension to place the 3 elements in.
+            If `None`, place the results as list elements in an object DataArray
+        sep : str, default `" "`
+            String to split on.
+
+        Returns
+        -------
+        partitioned : same type as values or object array
+
+        See Also
+        --------
+        DataArray.str.rpartition
+        str.partition
+        pandas.Series.str.partition
+        """
+        return self._partitioner(func=self._obj.dtype.type.partition, dim=dim, sep=sep)
+
+    def rpartition(
+        self,
+        dim: Optional[Hashable],
+        sep: Union[str, bytes] = " ",
+    ) -> Any:
+        """
+        Split the strings in the DataArray at the last occurrence of separator `sep`.
+
+        This method splits the string at the last occurrence of `sep`,
+        and returns 3 elements containing the part before the separator,
+        the separator itself, and the part after the separator.
+        If the separator is not found, return 3 elements containing two empty strings,
+        followed by the string itself.
+
+        This is equivalent to :meth:`str.rpartion`.
+
+        Parameters
+        ----------
+        dim : Hashable or `None`
+            Name for the dimension to place the 3 elements in.
+            If `None`, place the results as list elements in an object DataArray
+        sep : str, default `" "`
+            String to split on.
+
+        Returns
+        -------
+        rpartitioned : same type as values or object array
+
+        See Also
+        --------
+        DataArray.str.partition
+        str.rpartition
+        pandas.Series.str.rpartition
+        """
+        return self._partitioner(func=self._obj.dtype.type.rpartition, dim=dim, sep=sep)
+
+    def _splitter(
+        self,
+        func: Callable,
+        pre: bool,
+        dim: Hashable,
+        sep: Optional[Union[str, bytes]],
+        maxsplit: int,
+    ) -> Any:
+        """
+        Implements logic for `split` and `rsplit`.
+        """
+        if sep is not None:
+            sep = self._stringify(sep)
+
+        if dim is None:
+            f = lambda x: func(x, sep, maxsplit)
+            return self._apply(f, dtype=np.object_)
+
+        # _apply breaks on an empty array in this case
+        if not self._obj.size:
+            return self._obj.copy().expand_dims({dim: 0}, -1)
+
+        f_count = lambda x: max(len(func(x, sep, maxsplit)), 1)
+        maxsplit = self._apply(f_count, dtype=np.int_).max().data.tolist() - 1
```

Comment:
is the `tolist` necessary here? As far as I can tell a bare `max` will always return a `numpy` scalar (wrapped by a `xarray` object). If it is, using `.item()` might be cleaner.

Reply:
That seems to work.